From d397060684b09f4d0762aadf44d2a56f55510903 Mon Sep 17 00:00:00 2001
From: Nigel Croxon <ncroxon@redhat.com>
Date: Thu, 20 Mar 2014 18:43:42 -0400
Subject: [watchdog] hpwdt: display informative string

Message-id: <1395341022-7002-1-git-send-email-ncroxon@redhat.com>
Patchwork-id: 78463
O-Subject: [RHEL7.1 PATCH 1/1] watchdog: hpwdt patch to display informative string
Bugzilla: 1074038
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Thomas Mingarelli <thomas.mingarelli@hp.com>

Addresses Bugzilla 1074038
https://bugzilla.redhat.com/show_bug.cgi?id=1074038

Brew build 7212416
https://brewweb.devel.redhat.com/taskinfo?taskID=7212416

Tested by Lisa Mitchell & Randy Schneider at HP
Installed tested kernel, and did a TOC dump on a 3 blade system
and it paniced with the correct new panic message.

commit 7bb5be947e5a32f3b4c1c5cad3bf47e9b5a26db0
Author: Thomas Mingarelli <thomas.mingarelli@hp.com>
Date:   Tue Jan 28 21:26:10 2014 +0100
watchdog: hpwdt patch to display informative string

This patch is being submitted to output a general string when the panic comes
in that informs the user of the possible places to look for the source of the
NMI. Because various systems log the message in different places this would
give a single display of where to go look instead of code that acts on all
these different server names or IDs.

Signed-off-by: Thomas Mingarelli <thomas.mingarelli@hp.com>
Signed-off-by: Wim Van Sebroeck <wim@iguana.be>
(cherry picked from commit 7bb5be947e5a32f3b4c1c5cad3bf47e9b5a26db0)

Signed-off-by: Nigel Croxon <ncroxon@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/watchdog/hpwdt.c b/drivers/watchdog/hpwdt.c
index 03f2a46..f57e5a2 100644
--- a/drivers/watchdog/hpwdt.c
+++ b/drivers/watchdog/hpwdt.c
@@ -39,7 +39,7 @@
 #endif /* CONFIG_HPWDT_NMI_DECODING */
 #include <asm/nmi.h>
 
-#define HPWDT_VERSION   "1.3.2"
+#define HPWDT_VERSION   "1.3.3"
 #define SECS_TO_TICKS(secs)  ((secs) * 1000 / 128)
 #define TICKS_TO_SECS(ticks)  ((ticks) * 128 / 1000)
 #define HPWDT_MAX_TIMER   TICKS_TO_SECS(65535)
@@ -499,8 +499,13 @@ static int hpwdt_pretimeout(unsigned int ulReason, struct pt_regs *regs)
     "but unable to determine source.\n");
   }
  }
- panic("An NMI occurred, please see the Integrated "
-  "Management Log for details.\n");
+ panic("An NMI occurred. Depending on your system the reason "
+  "for the NMI is logged in any one of the following "
+  "resources:\n"
+  "1. Integrated Management Log (IML)\n"
+  "2. OA Syslog\n"
+  "3. OA Forward Progress Log\n"
+  "4. iLO Event Log");
 
 out:
  return NMI_DONE;
-- 
1.7.1